File: src/coding_problems/misc/CheckSubstr.java
package coding_problems.misc;

class CheckSubstr {
    static int checkSubstr(String s, String sub) {
        for (int i = 0; i < s.length() - sub.length() + 1; i++) {
            boolean flag = true;
            for (int j = 0; j < sub.length(); j++) {
                if (s.charAt(i + j) != sub.charAt(j)) {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                return i;
            }
        }

        return -1;
    }

    public static void main(String[] args) {
        System.out.println(checkSubstr("abcdefg", "bc"));
        System.out.println(checkSubstr("abcdefg", "ac"));
        System.out.println(checkSubstr("abcdefg", "a"));
        System.out.println(checkSubstr("abcdefg", "fg"));
        System.out.println(checkSubstr("abcdefg", "abcdefg"));
    }
}

File: src/coding_problems/misc/check_substr.py
def check_substr(s, sub):
    print s, sub
    length = len(sub)
    for i in range(len(s)):
        if s[i: i + length] == sub:
            return i

    return -1

if __name__ == '__main__':
    print check_substr('abcdefg', 'abc')
    print check_substr('abcdefg', 'abcdefg')
    print check_substr('abcdefg', 'fg')
    print check_substr('abcdefg', 'ffg')
    print check_substr('abcdefg', 'cdf')
    print check_substr('abcdefg', 'cdef')
    print check_substr('abcdefg', 'a')
    print check_substr('abcdefg', 'g')
